File: software/examples/csharp/ExampleSetSegments.cs
using System;
using Tinkerforge;

class Example
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string UID = "XYZ"; // Change XYZ to the UID of your Segment Display 4x7 Bricklet 2.0

	static void Main()
	{
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletSegmentDisplay4x7V2 sd =
		  new BrickletSegmentDisplay4x7V2(UID, ipcon); // Create device object

		ipcon.Connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		sd.SetBrightness(7); // Set to full brightness

		// Activate all segments
		sd.SetSegments(new bool[]{true, true, true, true, true, true, true, true},
		               new bool[]{true, true, true, true, true, true, true, true},
		               new bool[]{true, true, true, true, true, true, true, true},
		               new bool[]{true, true, true, true, true, true, true, true},
		               new bool[]{true, true}, true);

		Console.WriteLine("Press enter to exit");
		Console.ReadLine();
		ipcon.Disconnect();
	}
}

File: software/examples/rust/example_set_segments.rs
use std::{error::Error, io};

use tinkerforge::{ip_connection::IpConnection, segment_display_4x7_v2_bricklet::*};

const HOST: &str = "localhost";
const PORT: u16 = 4223;
const UID: &str = "XYZ"; // Change XYZ to the UID of your Segment Display 4x7 Bricklet 2.0.

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection.
    let sd = SegmentDisplay4x7V2Bricklet::new(UID, &ipcon); // Create device object.

    ipcon.connect((HOST, PORT)).recv()??; // Connect to brickd.
                                          // Don't use device before ipcon is connected.

    sd.set_brightness(7); // Set to full brightness

    // Activate all segments
    sd.set_segments(
        [true, true, true, true, true, true, true, true],
        [true, true, true, true, true, true, true, true],
        [true, true, true, true, true, true, true, true],
        [true, true, true, true, true, true, true, true],
        [true, true],
        true,
    );

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;
    ipcon.disconnect();
    Ok(())
}
